import.go

```diff
@@ -17,55 +19,78 @@ import (
 // @Produce  json
 // @Param import body main.Import _ "import object"
 // @Success 201 "Successfully imported"
-// @Failure 400 {object} string "Bad request"
-// @Failure 401 {object} string "Unauthorized or missing jwt token"
+// @Failure 400 {object} problem.Response "Bad request"
+// @Failure 401 {object} problem.Response "Unauthorized or missing jwt token"
+// @Failure 502 {object} problem.Response "Bad gateway"
 // @Router /gitlab [post]
 func runGitLabHandler(c *gin.Context) {
 	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
 
 	i := Import{}
 	err := c.BindJSON(&i)
-	if handleError(c, err) {
+	if err != nil {
+		ginutil.WriteInvalidBindError(c, err,
+			"One or more parameters failed to parse when reading the request body for GitHub projects import/refresh")
 		return
 	}
 
 	importer, err := newGitLabImporter(c.GetHeader("Authorization"), &i)
-	if handleError(c, err) {
+	if err != nil {
+		if !handleIfAuthError(c, err) {
+			ginutil.WriteProblemError(c, err, problem.Response{
+				Type: "prob/provider/gitlab/creating-importer-error",
+				Title: "Error creating GitLab importer.",
+				Status: http.StatusBadRequest,
+				Detail: "Creation of GitLab importer failed. Check your settings and make sure they are correct.",
+			})
+		}
+
```

Comment:
Looking at the implementation, this ought to be an API read error, right?

Suggest moving this problem handling into the newGitLabImporter, similar to how you've done it in the azuredevops PR https://github.com/iver-wharf/wharf-provider-azuredevops/pull/14 with the `WritesError` functions there.

Reply:
It's both Read and Write.
Will move the writing of the problems to the functions called in newGitLabImporter 👍🏻